pkg/context/tcpcontext.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package context
+
+type (
+	// TCPContext is context for TCP protocol
+	TCPContext interface {
+		Context
+
+		// TODO
```

Comment:
Is this TODO up-to-date? If nothing missing, remove this.

Reply:
the new pipeline not support TCP protocol yet. we will add more protocol later....

Reply:
```suggestion
		// TODO more protocols coming soon
```